src/vault/client/mock.clj

```diff
@@ -0,0 +1,226 @@
+(ns vault.client.mock
+  (:require
+    [clojure.edn :as edn]
+    [clojure.java.io :as io]
+    [clojure.string :as str]
+    [clojure.tools.logging :as log]
+    [vault.core :as vault])
+  (:import
+    java.text.SimpleDateFormat
+    java.net.URI
+    (java.util
+      Date
+      UUID)))
+
+
+(defn- gen-date
+  "Generates a formatted date-time string for the current instant."
+  []
+  (let [f (java.text.SimpleDateFormat. "yyyy-MM-DDHH:mm:ss.SSSZ")]
+    (.format f (Date.))))
+
+
+(defn- gen-uuid
+  "Generates a random UUID string."
+  []
+  (str (UUID/randomUUID)))
+
+
+(defn- mock-token-auth
+  "Generates a mock token response for use in the mock client."
+  []
+  {:client-token (gen-uuid)
+   :accessor (gen-uuid)
+   :policies ["root"]
+   :metadata nil
+   :lease-duration 0
+   :renewable false})
+
+
+(defrecord MockClient
+  [auth memory cubbies]
+
+  vault/Client
+
+  (authenticate!
+    [this auth-type credentials]
+    this)
+
+  (status
+    [this]
+    {:initialized true
+     :sealed false
+     :standby false
+     :server-time-utc (/ (System/currentTimeMillis) 1000)})
+
+
+  vault/TokenManager
+
+  (create-token!
+    [this opts]
+    {:request-id ""
+     :lease-id ""
+     :renewable false
+     :lease-duration 0
+     :data nil
+     :wrap-info
+     (when (:wrap-ttl opts)
+       (let [wrap-token (gen-uuid)]
+         (swap! cubbies assoc wrap-token (mock-token-auth))
+         {:token wrap-token
+          :ttl (:wrap-ttl opts)
+          :creation-time (gen-date)
+          :wrapped-accessor (gen-uuid)}))
+     :warnings nil
+     :auth (when-not (:wrap-ttl opts) (mock-token-auth))})
+
+  (lookup-token
+    [this]
+    ; TODO: implement
```

Comment:
Empty function bodies evaluate to `nil` already. That said, these definitely need some love, which I'll do as part of a separate PR.